### PR TITLE
Rsdk-8155: adding sats in use to sensor reading

### PR DESCRIPTION
**Pull request by susmitaSanyal**
Turns out we were using `sats_in_view `wrong. `sats_in_use` is used to calculate position and measure fix type. `sats_in_view` is the number of satellites visible to the receiver and it may or may not use it for positioning. 
I added some comments in the code to clarify the difference between the two and surfacing `sats_in_use` in sensor reading which is what we need to determine the fix type

### CODE REVIEW

**Review by penguinland — APPROVED** [review]
LGTM.

Why would these numbers ever differ? Phrased another way, why would there ever be satellites we can see that we would choose not to use?

**Comment by susmitaSanyal** [comment]
> Why would these numbers ever differ? Phrased another way, why would there ever be satellites we can see that we would choose not to use?

@penguinland few reasons can be proximity or if the receiver not configured to use a constellation

File: components/movementsensor/gpsnmea/component.go [hunk 1]
```diff
@@ -96,6 +96,11 @@ func (g *NMEAMovementSensor) ReadSatsInView(ctx context.Context) (int, error) {
 	return g.cachedData.ReadSatsInView(ctx)
 }
 
+// ReadSatsInUse returns the number of satellites in use.
+func (g *NMEAMovementSensor) ReadSatsInUse(ctx context.Context) (int, error) {
+	return g.cachedData.ReadSatsInUse(ctx)
+}
+
```
Comment (randhid) [inline components/movementsensor/gpsnmea/component.go:103]: Not for this pr, but could we add following helper in gpsutils:
`map, err = g.CachedData.GetCommonReadings` -> outputs Fix, Sats, all the exported functions that we stuff into just readings for the nmea movement sensor.

That way we don't have a weird  dependency of a private functions in gpsrtk -> public functions in gpsnmea/component.go -> public functions in gpsutils/cachedData.go.

In my opinion, the `ReadFix`, `ReadSatsInView`, and `ReadSatsInUse` helpers should disappear, both from the nmeaSensor interface and the Readings function in both drivers. Just get the data from the cache, we already parse and store it there.

**Comment by randhid** [comment]
> LGTM.
> 
> Why would these numbers ever differ? Phrased another way, why would there ever be satellites we can see that we would choose not to use?

Not all satellites provide GPS data, they have to be in specific orbits with specific clocks, but all satellites can be "seen" by GPS antennas. 

A good example of such satellites not in GNSS constellations are Starlink, where NONE of the thousands of them provide any GPS receiver on earth with any positioning data.

**Pull request merged**